src/legacy/ui/field_formats/types.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export type ContentType = 'html' | 'text';
+
+export interface IFieldFormat {
```

Comment:
// brain-pick

Could we benefit from using generics here?

Reply:
Not sure that we should use generics here.

Please see: 
![image](https://user-images.githubusercontent.com/20072247/64181423-19241e80-ce6f-11e9-8f85-307d3290e944.png)